contracts/pcv/tokemak/EthTokemakPCVDeposit.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "./TokemakPCVDepositBase.sol";
+import "../../Constants.sol";
+
+interface ITokemakEthPool {
+    function deposit(uint256 amount) external payable;
+    function withdraw(uint256 requestedAmount, bool asEth) external;
+}
+
+/// @title ETH implementation for a Tokemak PCV Deposit
+/// @author Fei Protocol
+contract EthTokemakPCVDeposit is TokemakPCVDepositBase {
+
+    /// @notice Tokemak ETH PCV Deposit constructor
+    /// @param _core Fei Core for reference
+    /// @param _pool Tokemak pool to deposit in
+    /// @param _rewards Tokemak rewards contract
+    constructor(
+        address _core,
+        address _pool,
+        address _rewards
+    ) TokemakPCVDepositBase(_core, _pool, _rewards) {}
+
+    receive() external payable {}
+
+    /// @notice deposit ETH to Tokemak
+    function deposit()
+        external
+        override
+        whenNotPaused
+    {
+        uint256 amount = address(this).balance;
+
+        token.approve(pool, amount);
```

Comment:
What's this approval do/for?

Reply:
removed, WETH does not need approval (pure ETH is passed to the payable function), good catch :)